x-pack/plugins/apm/public/components/shared/FilterBar/DatePicker.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import datemath from '@elastic/datemath';
+import {
+  CommonProps,
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps
+} from '@elastic/eui';
+import { UndefinedType } from 'io-ts';
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { updateTimePicker } from 'x-pack/plugins/apm/public/store/urlParams';
+import {
+  fromQuery,
+  getQueryWithRisonParams,
+  risonSafeDecode,
+  toQuery
+} from '../Links/url_helpers';
+
+interface Props extends RouteComponentProps {
+  dispatchUpdateTimePicker: typeof updateTimePicker;
+}
+
+interface State {
+  from: string;
+  to: string;
+  isPaused: boolean;
+  refreshInterval: number;
+}
+
+interface QueryG {
+  time?: {
+    from: string;
+    to: string;
+  };
+  refreshInterval?: {
+    pause: boolean;
+    value: number;
+  };
+}
+
+declare module '@elastic/eui' {
+  interface OnTimeChangeProps {
+    start: string;
+    end: string;
+    isInvalid: boolean;
+    isQuickSelection: boolean;
+  }
+
+  interface OnRefreshChangeProps {
+    isPaused: boolean;
+    refreshInterval: number;
+  }
+
+  export type EuiSuperDatePickerProps = CommonProps & {
+    start: string;
+    end: string;
+    isPaused: boolean;
+    refreshInterval: number;
+    onTimeChange: (props: OnTimeChangeProps) => void;
+    onRefreshChange: (props: OnRefreshChangeProps) => void;
+  };
+  export const EuiSuperDatePicker: React.SFC<EuiSuperDatePickerProps>;
+}
+
+const APM_DEFAULT_TIME_RANGE = {
+  from: 'now-24h',
+  to: 'now'
+};
+
+const APM_DEFAULT_REFRESH = {
+  pause: true,
+  value: 0
+};
+
+class DatePickerComponent extends React.Component<Props, State> {
+  public state = {
+    from: '',
+    to: '',
+    isPaused: true,
+    refreshInterval: 0
+  };
```

Comment:
Why is this component stateful? Shouldn't it just read the props from urlParams (where we can also set the defaults)?

Reply:
That's possibly true, the state was just set up to get the EUI component to display correctly since it's controlled from here. But the query state may be enough to keep it synced, I'll take a look.

Reply:
Ah right, while we're still using rison, we need to store the decoded values in state or else decode them in every render.

Reply:
> Ah right, while we're still using rison, we need to store the decoded values in state or else decode them in every render.

Did you find any perf issues? Else it sounds a little like premature optimisation. 
I hope we can get rid of Rison, and even if we can't I think we should keep the parsing of url params consistent with all the other url params.

Reply:
Hm ok. Not sure about premature optimization, it just seems a bit weird to do the decode in every render and also in every refresh cycle. I’ll wait until we make a final decision on rison to worry about this though. 

> I think we should still keep the parsing of url params consistent with all the other url params - and then we can apply memoization where needed.

Do you mean by storing them in the redux store? We don't currently store any of this local datepicker-specific info in redux -- `urlParams.start` and `urlParams.end` are the calculated date values used by all the ES queries, but these date values need to be the raw strings found in `from` and `to` so they can be passed into the date picker component that way. We don't use those raw strings anywhere else in the app, and we don't use the pause and refresh interval state anywhere else either, fwiw. 

Once we sort the rison question I'll revisit this and see if we can remove the local state.

Reply:
> We don't currently store any of this local datepicker-specific info in redux

No, but don't you think that's a odd when we handle the other url params in the reducer?

Reply:
> Once we sort the rison question I'll revisit this and see if we can remove the local state

What do you think about trying it out, and seeing if it simplifies things, and then make a decision?

Reply:
> don't you think that's odd when we handle the other url params in the reducer?

I get what you're saying here re: consistency, but it also seems odd to move things into Redux that are only ever used in this one local component, and a bit complicated to manage `urlParams.start` next to `urlParams.from` etc. Not sure which odd thing is better here… let me think on it and we can chat in stand up.